internal/controller/helmchart_controller.go

```diff
@@ -665,8 +665,10 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 	// Construct the chart builder with scoped configuration
 	cb := chart.NewRemoteBuilder(chartRepo)
 	opts := chart.BuildOptions{
-		ValuesFiles: obj.GetValuesFiles(),
-		Force:       obj.Generation != obj.Status.ObservedGeneration,
+		ValuesFiles:              obj.GetValuesFiles(),
+		CachedChartValuesFiles:   obj.Status.ObservedValuesFiles,
```

Comment:
I believe it should changed here too, in order to be done in the `artifact if case`.

Reply:
Updated.